Extractor error

HFAsif opened the pull request:
P

LinPolly committed:
Fix Signature

LinPolly committed:
Update Decipherer.cs

LinPolly committed:
Update Program.cs

LinPolly committed:
Update Program.cs

LinPolly committed:
Incomplete upload

LinPolly committed:
U

LinPolly committed:
U

LinPolly committed:
Update .gitignore

LinPolly committed:
D

LinPolly committed:
U

iamtorsten committed:
Fix url for non encrypted urls

When a video do not need a decryption, we only need to normalize the base url and should not try get a signature. For case requiresDecryption was false, it was thrown an error.

LinPolly committed:
Fix non encrypted urls

LinPolly committed:
Update DownloadUrlResolver.cs

Remove unwanted symbols

iamtorsten committed:
Test decipherer -> seems broken

iamtorsten committed:
Fix decipher

LinPolly committed:
Merge pull request #2 from mrklintscher/master

Fix decipher

LinPolly committed:
Update DownloadUrlResolver.cs

LinPolly committed:
Fix jsUrl

LinPolly committed:
Support age restriction

HFAsif reviewed (commented):
i am getting some error to compile the project YoutubeModel and Formate missing

HFAsif commented:
can you please provide the two classes plss

LinPolly commented:
[https://github.com/LinPolly/YoutubeExtractor/blob/master/YoutubeExtractor/YoutubeExtractor/YoutubeModel.cs](https://github.com/LinPolly/YoutubeExtractor/blob/master/YoutubeExtractor/YoutubeExtractor/YoutubeModel.cs)

developerss commented:
that is great not everything works perfect . 

in the past i was use this code to get age restricted video 

private static IEnumerable<ExtractionInfo> ExtractDownloadUrls2(string pagecontent)
        {

            var orient = HttpHelper.ParseQueryString2(pagecontent);
            string getitems = orient["url_encoded_fmt_stream_map"];
            string adaptipmap = string.Empty;
            try
            {
                adaptipmap = orient["adaptive_fmts"];
            }
            catch { }

            string[] splitByUrls = getitems.Split(',');
            string[] adaptiveFmtSplitByUrls = adaptipmap.Split(',');


            splitByUrls = splitByUrls.Concat(adaptiveFmtSplitByUrls).ToArray();

            
            foreach (string s in splitByUrls)
            {
                if(s != null)
                {
                    if (s != string.Empty)
                    {
                        IDictionary<string, string> queries = HttpHelper.ParseQueryString(s);
                        string url;

                        bool requiresDecryption = false;

                        if (queries.ContainsKey("s") || queries.ContainsKey("sig"))
                        {
                            requiresDecryption = queries.ContainsKey("s");
                            string signature = queries.ContainsKey("s") ? queries["s"] : queries["sig"];

                            url = string.Format("{0}&{1}={2}", queries["url"], SignatureQuery, signature);

                            string fallbackHost = queries.ContainsKey("fallback_host") ? "&fallback_host=" + queries["fallback_host"] : String.Empty;

                            url += fallbackHost;
                        }

                        else
                        {
                            url = queries["url"];


                            try
                            {
                                try
                                {
                                    getquality = queries["quality"];
                                    try
                                    {
                                        getfps = queries["fps"];
                                    }
                                    catch { }
                                    //if (getquality == "1080p60")
                                    //{
                                    //    getfps = "60 fps";
                                    //}
                                    //else
                                    //{
                                    //    getfps = "";
                                    //}
                                }
                                catch
                                {
                                    getquality = queries["quality_label"];
                                }
                                try
                                {
                                    gettypes = queries["type"];
                                    MatchCollection m1 = Regex.Matches(gettypes, "video/\\s*(.+?)\\s*;");
                                    if (m1.Count > 0)
                                    {
                                        foreach (Match i in m1)
                                        {
                                            gettypes = i.Groups[1].Value;
                                        }
                                    }

                                }
                                catch { }
                                try
                                {
                                    getbirate = Convert.ToUInt32(queries["bitrate"]);
                                }
                                catch { }

                                try
                                {
                                    getsize = queries["size"];
                                }
                                catch { }

                            }
                            catch { }

                        }

                        url = HttpHelper.UrlDecode(url);
                        url = HttpHelper.UrlDecode(url);


                        IDictionary<string, string> parameters = HttpHelper.ParseQueryString(url);
                        if (!parameters.ContainsKey(RateBypassFlag))
                            url += string.Format("&{0}={1}", RateBypassFlag, "yes");


                        yield return new ExtractionInfo
                        {
                            RequiresDecryption = requiresDecryption,
                            Uri = new Uri(url),
                            GetvideoBirate = getbirate,
                            Getvideoquality = getquality,
                            Getvideosize = getsize,
                            GetVideoType = gettypes,
                            GetFps = getfps
                        };
                    }
                }
                
                
            }
        }


but now my code not working anymore

HFAsif commented:
many thanks to you for this awesome fix ,

HFAsif commented:
Please watch this image
https://imgur.com/QHQp8Bv

HFAsif commented:
i wish youtube portal don't update their strategy soon

LinPolly committed:
Fix GetHtml5PlayerVersion Regex pattern

LinPolly committed:
Fix Json To YoutubeModel & Update VideoInfo

LinPolly committed:
Fix by can't embed video

LinPolly committed:
Update regex

reference from youtube-dl youtube.py
agegate from #L1647
player_response from #L1696

LinPolly committed:
Update DownloadUrlResolver.cs

HFAsif committed:
an exception

if (!extractedJson.Contains("raw_player_response:ytInitialPlayerResponse")) //https://www.youtube.com/watch?v=9Y7TRMISkGE

LinPolly committed:
Merge pull request #5 from developerss/patch-3

an exception

HFAsif committed:
player respond should go correct method 

so i used a try catch to go second method ,

HFAsif committed:
you can fix it better way

i wish it

LinPolly committed:
Merge pull request #6 from developerss/patch-4

player respond should go correct method

LinPolly committed:
Revert "Merge pull request #6 from developerss/patch-4"

This reverts commit b07db428c53e549fed09087b94a0093149161d29, reversing
changes made to 34a99d47ef8456dff0b8a4928c3af947b146e676.

LinPolly committed:
Reference youtube-dl content modification

LinPolly committed:
Fix download slow

Decipherer Fix From
yt_dlp/extractor/youtube.py#L1882
yt_dlp/jsinterp.py#L489

LinPolly committed:
clean unused

LinPolly committed:
Update Decipherer.cs

Fix Dev Pattern
